docs/src/design/many_differentials.md

```diff
@@ -0,0 +1,188 @@
+
+So the structural differential is another type of differential.
+We must support both natural and structual differentials, because AD can only create structural differentials, unless using custom sensitivity rules, and all custom sensitivities are only written in terms of natural differentials, as that is what is used in papers about derivatives.
+
+## Semi-structural differentials
+
+Where there is no natural differential type for the outermost type, but there is for some of its fields, which can call this a semi-structural differential.
+
+Consider if we had a representation of a country's GDP as output by some continuous time model like a Gaussian Process, where that representation is as a sequence of `TimeSample`s.
+
+Where a `TimeSample` is a structure as follows:
+```julia
+julia> struct TimeSample
+           time::DateTime
+           value::Float64
+       end
+```
+
+We can look at its structure:
+```julia
+julia> dump(TimeSample(now(), 2.6e9))
+TimeSample
+  time: DateTime
+    instant: Dates.UTInstant{Millisecond}
+      periods: Millisecond
+        value: Int64 63720043490844
+  value: Float64 2.6e9
+```
+
+Thus we see the that structural differential would be:
+```julia
+Composite{TimeSample}(
+    time::Composite{DateTime}(
+        instant::Composite{UTInstant{Millisecond}}(
+            periods::Composite{Millisecond}(
+                value::Int64
+            )
+        )
+    ),
+    value::Float64
+)
+```
+
+But instead in the custom sensitivity rule we would write a semi-structured differential type.
+Since there is not a natural differential type for `TimeSample` but there is for `DateTime`.
+```julia
+Composite{TimeSample}(
+    time::Day,
+    value::Float64
+)
+```
+
+So the rule author has written a structural differential with some fields that are natural differentials.
+
+Another related case is for types that overload `getproperty` such as `SVD` and `QR`.
+In this case the structural differential will be based on the fields, but those fields do not always have an easy relation to what is actually used in math.
+For example, the `QR` type has fields `factors` and `t`, but we would more naturally think in terms of the properties `Q` and `R`.
+So most rule authors would want to write semi-structural differentials based on the properties.
+
+To return to the question of why ChainRules has `Composite{P, <:NamedTuple}` whereas Zygote v0.4 just has `NamedTuple`, it relates to semi-structural derivatives, and being able to overload things more generally.
+If one knows that one has a semi-structural derivative based on property names, like `Composite{QR}(Q=..., R=...)`, and one is adding it to the true structural derivative based on field names `Composite{QR}(factors=..., τ=...)`, then we need to overload the addition operator to perform that correctly.
+We cannot overload happily similar things for `NamedTuple` since we don't know the primal type, only the names of the values contained.
```

Comment:
```suggestion
We cannot happily overload similar things for `NamedTuple` since we don't know the primal type, only the names of the values contained.
```